domain/cloud/service/service.go

```diff
@@ -24,7 +23,7 @@ type State interface {
 	Save(context.Context, cloud.Cloud) error
 
 	// List returns the clouds matching the optional filter.
-	List(context.Context, *state.Filter) ([]cloud.Cloud, error)
+	List(context.Context, string) ([]cloud.Cloud, error)
```

Comment:
I think we should just call it `ListByName` and then we can build on the convention with `ListCloudsByName`.

Reply:
It's not really a *list* by name, since there's only one. I don't mind this, because you would have to push down re-use somewhere.

At the the service layer, it's `Get` and `List`, so the API we'll consume remains clean - just some mechanics at the state layer.